Nil pointer dereference around execStagingOperation

We recently upgraded to 1.7.0 and started getting a `nil` pointer. We downgraded to 1.6.1, and things are stable now. Corresponding stack trace:
```GO
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2642668]

goroutine 5829 [running]:
github.com/databricks/databricks-sql-go.(*conn).execStagingOperation(0x4017f8b170, 0x4000db1d50, {0x47db718, 0x4017da5260})
	/go/pkg/mod/github.com/databricks/databricks-sql-go@v1.7.0/connection.go:584 +0xd8
github.com/databricks/databricks-sql-go.(*conn).ExecContext(0x4017f8b170, {0x47db7c0?, 0x40007d6690?}, {0x40013c6f60, 0x1a}, {0x4000f8f0c8, 0x0, 0x0})
	/go/pkg/mod/github.com/databricks/databricks-sql-go@v1.7.0/connection.go:119 +0x15c
github.com/databricks/databricks-sql-go.(*connector).Connect(0x4003e91ca0, {0x47db7c0, 0x40007d6690})
	/go/pkg/mod/github.com/databricks/databricks-sql-go@v1.7.0/connector.go:70 +0x798
database/sql.(*DB).conn(0x4014efd450, {0x47db7c0, 0x40007d6690}, 0x1)
	/usr/local/go/src/database/sql/sql.go:1431 +0x9b0
database/sql.(*DB).query(0x4014efd450, {0x47db7c0, 0x40007d6690}, {0x401aa5ebe0, 0x43}, {0x0, 0x0, 0x0}, 0x0?)
	/usr/local/go/src/database/sql/sql.go:1759 +0x40
database/sql.(*DB).QueryContext.func1(0xbb?)
	/usr/local/go/src/database/sql/sql.go:1742 +0x54
database/sql.(*DB).retry(0x4000f8f658?, 0x4000f8f630)
	/usr/local/go/src/database/sql/sql.go:1576 +0x4c
database/sql.(*DB).QueryContext(0x47db718?, {0x47db7c0?, 0x40007d6690?}, {0x401aa5ebe0?, 0x4001d65b30?}, {0x0?, 0x1?, 0x1?})
	/usr/local/go/src/database/sql/sql.go:1741 +0x9
```

--- a/connection.go
+++ b/connection.go
@@ -571,8 +571,8 @@ func (c *conn) execStagingOperation(
 	var err error
 
 	var isStagingOperation bool
-	if exStmtResp.DirectResults != nil && exStmtResp.DirectResults.ResultSetMetadata != nil && exStmtResp.DirectResults.ResultSetMetadata.IsStagingOperation != nil {
-		isStagingOperation = *exStmtResp.DirectResults.ResultSetMetadata.IsStagingOperation
+	if exStmtResp.DirectResults != nil && exStmtResp.DirectResults.ResultSetMetadata != nil {
+		isStagingOperation = exStmtResp.DirectResults.ResultSetMetadata.IsStagingOperation != nil && *exStmtResp.DirectResults.ResultSetMetadata.IsStagingOperation
 	} else {
 		req := cli_service.TGetResultSetMetadataReq{
 			OperationHandle: exStmtResp.OperationHandle,
@@ -581,7 +581,7 @@ func (c *conn) execStagingOperation(
 		if err != nil {
 			return dbsqlerrint.NewDriverError(ctx, "error performing staging operation", err)
 		}
-		isStagingOperation = *resp.IsStagingOperation
+		isStagingOperation = resp.IsStagingOperation != nil && *resp.IsStagingOperation
 	}
 
 	if !isStagingOperation {

--- a/connection_test.go
+++ b/connection_test.go
@@ -1677,6 +1677,100 @@ func TestConn_PrepareContext(t *testing.T) {
 	})
 }
 
+func TestConn_execStagingOperation(t *testing.T) {
+	t.Run("handles nil IsStagingOperation from DirectResults", func(t *testing.T) {
+		testClient := &client.TestClient{}
+		testConn := &conn{
+			session: getTestSession(),
+			client:  testClient,
+			cfg:     config.WithDefaults(),
+		}
+
+		// Create response with nil IsStagingOperation in DirectResults
+		exStmtResp := &cli_service.TExecuteStatementResp{
+			Status: &cli_service.TStatus{
+				StatusCode: cli_service.TStatusCode_SUCCESS_STATUS,
+			},
+			OperationHandle: &cli_service.TOperationHandle{
+				OperationId: &cli_service.THandleIdentifier{
+					GUID:   []byte{1, 2, 3, 4, 2, 23, 4, 2, 3, 1, 2, 3, 4, 223, 34, 54},
+					Secret: []byte("b"),
+				},
+			},
+			DirectResults: &cli_service.TSparkDirectResults{
+				ResultSetMetadata: &cli_service.TGetResultSetMetadataResp{
+					Status: &cli_service.TStatus{
+						StatusCode: cli_service.TStatusCode_SUCCESS_STATUS,
+					},
+					// IsStagingOperation is nil
+				},
+			},
+		}
+
+		// Mock GetResultSetMetadata to return false for IsStagingOperation
+		var getResultSetMetadataCount int
+		getResultSetMetadata := func(ctx context.Context, req *cli_service.TGetResultSetMetadataReq) (_r *cli_service.TGetResultSetMetadataResp, _err error) {
+			getResultSetMetadataCount++
+			var falseVal = false
+			return &cli_service.TGetResultSetMetadataResp{
+				Status: &cli_service.TStatus{
+					StatusCode: cli_service.TStatusCode_SUCCESS_STATUS,
+				},
+				IsStagingOperation: &falseVal,
+			}, nil
+		}
+		testClient.FnGetResultSetMetadata = getResultSetMetadata
+
+		ctx := context.Background()
+		err := testConn.execStagingOperation(exStmtResp, ctx)
+
+		assert.Nil(t, err)
+		assert.Equal(t, 0, getResultSetMetadataCount) // should not be called since DirectResults.ResultSetMetadata exists
+	})
+
+	t.Run("handles nil IsStagingOperation from GetResultSetMetadata", func(t *testing.T) {
+		testClient := &client.TestClient{}
+		testConn := &conn{
+			session: getTestSession(),
+			client:  testClient,
+			cfg:     config.WithDefaults(),
+		}
+
+		// Create response with nil DirectResults.ResultSetMetadata
+		exStmtResp := &cli_service.TExecuteStatementResp{
+			Status: &cli_service.TStatus{
+				StatusCode: cli_service.TStatusCode_SUCCESS_STATUS,
+			},
+			OperationHandle: &cli_service.TOperationHandle{
+				OperationId: &cli_service.THandleIdentifier{
+					GUID:   []byte{1, 2, 3, 4, 2, 23, 4, 2, 3, 1, 2, 3, 4, 223, 34, 54},
+					Secret: []byte("b"),
+				},
+			},
+			// DirectResults.ResultSetMetadata is nil
+		}
+
+		// Mock GetResultSetMetadata to return nil for IsStagingOperation
+		var getResultSetMetadataCount int
+		getResultSetMetadata := func(ctx context.Context, req *cli_service.TGetResultSetMetadataReq) (_r *cli_service.TGetResultSetMetadataResp, _err error) {
+			getResultSetMetadataCount++
+			return &cli_service.TGetResultSetMetadataResp{
+				Status: &cli_service.TStatus{
+					StatusCode: cli_service.TStatusCode_SUCCESS_STATUS,
+				},
+				// IsStagingOperation is nil
+			}, nil
+		}
+		testClient.FnGetResultSetMetadata = getResultSetMetadata
+
+		ctx := context.Background()
+		err := testConn.execStagingOperation(exStmtResp, ctx)
+
+		assert.Nil(t, err)
+		assert.Equal(t, 1, getResultSetMetadataCount) // should be called since DirectResults.ResultSetMetadata is nil
+	})
+}
+
 func getTestSession() *cli_service.TOpenSessionResp {
 	return &cli_service.TOpenSessionResp{SessionHandle: &cli_service.TSessionHandle{
 		SessionId: &cli_service.THandleIdentifier{

